File: java/src/Patch.java
package hermes;

public class Patch {
    public static String applyDiffBlock(String content, String searchBlock, String replaceBlock) throws Exception {
        int index = content.indexOf(searchBlock);
        if (index < 0) {
            throw new Exception("search block not found in file content");
        }

        return content.substring(0, index) + replaceBlock + content.substring(index + searchBlock.length());
    }
}

File: csharp/Patch.cs
using System;

namespace Hermes
{
    public static class Patch
    {
        public static string ApplyDiffBlock(string content, string searchBlock, string replaceBlock)
        {
            int index = content.IndexOf(searchBlock, StringComparison.Ordinal);
            if (index < 0)
            {
                throw new Exception("search block not found in file content");
            }

            return content.Substring(0, index) + replaceBlock + content.Substring(index + searchBlock.Length);
        }
    }
}
